src/tools/scripts/set_lighthouse_urls.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+LIGHTHOUSE_CONFIG_FILE="../.github/lighthouse/lighthouse-config-dev.json"
+LIGHTHOUSE_PROD_CONFIG_FILE="../.github/lighthouse/lighthouse-config-prod.json"
+
+# Usage info
+show_help() {
+cat << EOH
+Usage: ${0##*/} [-p]
+Get a list of URLs to run a lighthouse test on.
+If a commit is given then only URLs changes are tested otherwise all.
+
+    -h   display this help and exit
+    -p   get all production URLs from sitemap.xml
+EOH
+}
+
+OPTIND=1 # Reseting is good practive
+production=0
+while getopts "h?p" opt; do
+    case "$opt" in
+    h|\?)
+        show_help
+        exit 0
+        ;;
+    p)  production=1
+        ;;
+    esac
+done
+shift "$((OPTIND-1))" # Discard the options and sentinel --
+
+# This script must be run from src directory
+if [ -d "src" ]; then
+  cd src || exit
+fi
+
+LIGHTHOUSE_URLS=""
+
+# Set some URLs that should always be checked on pull requests
+# to ensure basic coverage
+BASE_URLS=$(cat <<-END
+http://127.0.0.1:8080/en/2019/
+http://127.0.0.1:8080/en/2019/javascript
+http://127.0.0.1:8080/en/2020/
+END
+)
+
+if [ "${production}" == "1" ]; then
+    
+    # Get the production URLs from the production sitemap
+    LIGHTHOUSE_URLS=$(curl -s https://almanac.httparchive.org/sitemap.xml | grep "<loc" | grep -v static | sed 's/ *<loc>//g' | sed 's/<\/loc>//g')
```

Comment:
Agree with changing `static` to `/static/`. Don't agree the other change is simpler! I'm having trouble getting my head round it to be honest and I think my unix is pretty decent!

Reply:
Explaining `grep -oP '<loc>\K\S+(?=</loc>)'`:

`-o`: Only output the matching part, not the whole line.
`-P`: Use Perl style RegEx to avoid unnecessary escaping and having access to more powerful means of capture groups.
`\K`: What follows this will be recorded as a capture group.
`\S`: Anything that is not a white-space character.
`(?=</loc>)`: A look ahead pattern to declare when to end the previous capture group.

The problem with multiple sed instances is that it is very difficult to ensure they will be working in the same proximity and position as each instance is acting independently. Though, that is not an issue here because we already know the kind of input we are dealing with. Moreover, invoking multiple programs many times can accumulate over time to increase the processing time.

Reply:
If you do not like the `\K` notation, you can use a combination of look behind and look ahead to isolate the point of interest like this:

```
grep -oP '(?<=<loc>)\S+(?=</loc>)'
```

Reply:
> I'm having trouble getting my head round it to be honest and I think my unix is pretty decent!

It is less about Unix knowledge and more about RegEx. As they say, [Regular Expressions: Now You Have Two Problems](https://blog.codinghorror.com/regular-expressions-now-you-have-two-problems/)! :laughing:

Reply:
Yes put having two simple `sed` commands with simple regexs, versus one complex `grep` command with complex, difficult to understand and maintain, regex using capture groups and look ahead patterns is not simplifying it it my point.

Using pipes and keeping commands simple is a huge part of Unix philosophy precisely because it keeps things simple.

So thanks for the feedback but going to change `static` to `/static/` but keep the rest as it is.